src/john.c

```diff
@@ -324,7 +324,7 @@ static void john_register_one(struct fmt_main *format)
 	/* Format disabled in john.conf */
 	if (!override_disable &&
 	    cfg_get_bool(SECTION_DISABLED, SUBSECTION_FORMATS,
-	                 format->params.label, 0)) {
+	                 (char*)format->params.label, 0)) {
```

Comment:
Instead of doing these casts, we ought to add `const` to the prototype of `cfg_get_bool`.

Reply:
Yes, and all other cfg_get_* functions. Will you do it in core first or should I?

Reply:
OK, I'll do it in core first. Let's just not forget to remove these casts.

Reply:
I've just added `const` to cfg_* and path_* functions in core.